miso-web/src/main/webapp/scripts/sample_hot.js

```diff
@@ -934,6 +947,22 @@ Sample.hot = {
     });
   },
   
+  /**
+   * Serial execution of an array of functions. 
+   * @param {Function[]} arrayOfFunctions - Each function takes a single callback function parameter.
+   */
+  serial: function(arrayOfFunctions) {
+	  index = 0;
+	  
+	  function serialCallBack() {
+		  if(index < (arrayOfFunctions.length)) {
+		     arrayOfFunctions[index++](serialCallBack);
```

Comment:
I find this mutation of state to be really difficult to reason about. Can this be something like:

```
fuction(aof) {
   var invokeNext = function(index) {
       if (index < aof.length) { aof[index](function() { invokeNext(index + 1);} ); }
   }
   invokeNext(0);
}